pwnlib/elf/elf.py

```diff
@@ -1258,6 +1258,37 @@ def search(self, needle, writable = False, executable = False):
                     break
                 yield (addr + offset + load_address_fixup)
                 offset += 1
+        if not segments:
+            if writable:
+                ko_check_segments = ".data"
+                text_filesz=0
+                rodata_filesz=0
+                for section in super().iter_sections():
+                    if section.name == ".text":
+                        text_filesz = section['sh_size']
+                    elif len(section.name)>=len(".rodata") and section.name[:len(".rodata")]==".rodata":
+                        rodata_filesz += section['sh_size']
+                addr = (text_filesz//PAGESIZE + 1 + rodata_filesz//PAGESIZE + 1)*PAGESIZE
+            elif executable:
+                ko_check_segments = ".text"
+                addr = 0
+            else:
+                # There may be other sections before .rodata, such as .note.gnu.build-id or .note.Linux
```

Comment:
It may be related to the kernel compilation script . I'm not sure they all work the same way under different kernel versions or options, so I've left the rodata part on hold.